### PR TITLE
slice/maps: move to stdlib

### PR DESCRIPTION
**Pull request by howardjohn**
This moves our usage to the stdlib insted of exp. Exp is experimental,
so the intent is to have everyone move to stdlib eventually.

The big change here is that SortFunc is now returning an int instead of
bool. This is kind of annoying for most of our cases. To work around
this, I did:

* Made a new `SortBy` which covers 90% of cases
* Made an example of how to properly sort by multiple fields

And moved all usages to one of those two.

I also included the linter rule to ensure we always use the right
packages.

Finally, the change made a real behavioral change -- we were
unintentionally sorting backwards in istioctl output; so I updated the
test to align with the new correct output.

### CODE REVIEW

**Review by GregHanson — APPROVED** [review]

File: pkg/slices/slices_test.go [hunk 1]
```diff
@@ -308,3 +311,75 @@ func TestDereference(t *testing.T) {
...
+type myStruct struct {
+	a, b, c, d string
+	n          int
+}
+
+func makeRandomStructs(n int) []*myStruct {
+	rand.Seed(42) // nolint: staticcheck
+	structs := make([]*myStruct, n)
+	for i := 0; i < n; i++ {
+		structs[i] = &myStruct{n: rand.Intn(n)}
```
Comment (GregHanson) [inline pkg/slices/slices_test.go:325]:
```suggestion
		structs[i] = &myStruct{n: rand.Intn(n)} // nolint: staticcheck
```

**Review by zirain — APPROVED** [review]

**Review by hanxiaop — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]
should we bump the go version in go.mod

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**